src/screens/CreatePost/index.js

```diff
@@ -492,26 +495,31 @@ const CreatePost = () => {
               params: {refresh: true}
             });
           }
-
-          Analytics.logEvent('create_post', {
-            id: 6,
-            newpost_reach: renderLocationString(geoList[geoSelect]),
-            newpost_privacy: listPrivacy[0].label,
-            num_images: 0,
-            added_poll: isPollShown,
-            topics_added: listTopic,
-            anon: typeUser,
-            predicted_audience: audienceEstimations
+        } else {
+          showMessage({
+            message: StringConstant.createPostFailedGeneralError,
+            type: 'danger'
           });
+        }
 
-          return true;
+        Analytics.logEvent('create_post', {
+          id: 6,
+          newpost_reach: renderLocationString(geoList[geoSelect]),
+          newpost_privacy: listPrivacy[0].label,
+          num_images: 0,
+          added_poll: isPollShown,
+          topics_added: listTopic,
+          anon: typeUser,
+          predicted_audience: audienceEstimations
+        });
+      } catch (e) {
+        if (__DEV__) {
+          console.log('CreatePost : ', e);
+          console.warn('retryCount', retryCount);
         }
-        throw new Error('Post failed');
-      } catch (error) {
-        console.warn('retryCount', retryCount);
         if (retryCount >= maxRetries) {
           showMessage({
-            message: 'Post keeps failing. Please check your internet connection.',
+            message: 'Failed to post. Please check your internet connection.',
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling here is good, as it provides feedback to the user when an error occurs. However, the error message could be more specific. Instead of a generic "Please check your internet connection" message, consider using the actual error message from the exception (`e.message`). This will provide more detailed information about what went wrong, which can be helpful for debugging.

```diff
-           showMessage({
-             message: 'Failed to post. Please check your internet connection.',
-             type: 'danger'
-           });
+          showMessage({
+            message: e.message,
+            type: 'danger'
+          });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->